ext/featurefmt/apk/apk.go

```diff
@@ -79,6 +81,11 @@ func (l lister) ListFeatures(files analyzer.Files) ([]database.FeatureVersion, e
 					Name:    pkg.Feature.Name,
 					Version: pkg.Version,
 				}
+
+				if source != "" {
```

Comment:
Doing it this way will ultimately end up with `ListFeatures` potentially returning multiple packages with the same name and version, but different `ExecutableToDependencies` and `LibraryToDependencies`. In this scenario, it is likely only one of those will be inserted into the database, so we may potentially end up with non-deterministic behavior when it comes to figuring out a package's provided executables and those executables' dependencies.

For this reason, we should update the `pkg.Feature.Name` to be `source` prior to adding it to the `pkgSet`. However, this now means we will only be using the `ExecutableToDependencies` and `LibraryToDependencies` from a single package. We should combine them via `Merge`.

You can try something like this:

```
if source != "" {
	pkg.Feature.Name = source
	source = "" // This can be moved outside the outer `if` where the other variables are reset
}

key := featurefmt.PackageKey{
	Name:    pkg.Feature.Name,
	Version: pkg.Version,
}
if oldPkg, exists := pkgSet[key]; exists {
	pkg.ExecutableToDependencies.Merge(oldPkg.ExecutableToDependencies)
	pkg.LibraryToDependencies.Merge(oldPkg.LibraryToDependencies)
}

pkgSet[key] = pkg
```